ingestion/src/metadata/ingestion/source/mssql_usage.py

```diff
@@ -0,0 +1,142 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+MSSQL usage module
+"""
+
+from typing import Any, Dict, Iterable, Iterator, Union
+
+from metadata.generated.schema.entity.services.databaseService import (
+    DatabaseServiceType,
+)
+from metadata.ingestion.api.source import Source, SourceStatus
+
+# This import verifies that the dependencies are available.
+from metadata.ingestion.models.table_queries import TableQuery
+from metadata.ingestion.ometa.openmetadata_rest import MetadataServerConfig
+from metadata.ingestion.source.mssql import MssqlConfig
+from metadata.ingestion.source.sql_alchemy_helper import (
+    SQLAlchemyHelper,
+    SQLSourceStatus,
+)
+from metadata.utils.helpers import get_start_and_end
+from metadata.utils.sql_queries import MSSQL_SQL_STATEMENT
+
+
+class MssqlUsageSource(Source[TableQuery]):
+    """
+    MSSQL Usage source
+
+    Args:
+        config:
+        metadata_config:
+        ctx:
+
+    Attributes:
+        config:
+        analysis_date:
+        sql_stmt:
+        alchemy_helper:
+        _extract_iter:
+        _database:
+        report:
+        SQL_STATEMENT (str):
+        WHERE_CLAUSE_SUFFIX_KEY (str):
+        CLUSTER_SOURCE (str):
+        USE_CATALOG_AS_CLUSTER_NAME (str):
+        DATABASE_KEY (str):
+        SERVICE_TYPE (str):
+        DEFAULT_CLUSTER_SOURCE (str):
+    """
+
+    # SELECT statement from mysql information_schema
+    # to extract table and column metadata
+    SQL_STATEMENT = MSSQL_SQL_STATEMENT
+
+    # CONFIG KEYS
+    WHERE_CLAUSE_SUFFIX_KEY = "where_clause"
+    CLUSTER_SOURCE = "cluster_source"
+    CLUSTER_KEY = "cluster_key"
+    USE_CATALOG_AS_CLUSTER_NAME = "use_catalog_as_cluster_name"
+    DATABASE_KEY = "database_key"
+    SERVICE_TYPE = DatabaseServiceType.MSSQL.value
+    DEFAULT_CLUSTER_SOURCE = "CURRENT_DATABASE()"
+
+    def __init__(self, config, metadata_config, ctx):
+        super().__init__(ctx)
+        self.config = config
+        start, end = get_start_and_end(config.duration)
+        self.analysis_date = start
+        self.sql_stmt = MssqlUsageSource.SQL_STATEMENT.format(
+            start_date=start, end_date=end
+        )
+        self.alchemy_helper = SQLAlchemyHelper(
+            config, metadata_config, ctx, "MSSQL", self.sql_stmt
```

Comment:
If possible, let's try to avoid hardcoding values. Here we could use `DatabaseServiceType.MSSQL.value` instead